lib/medtronic/medtronicSimulator.js

```diff
@@ -66,6 +77,111 @@ exports.make = function(config){
   }
 
   return {
+    basal: function(event) {
+      ensureTimestamp(event);
+
+      if (currBasal != null) {
+
+        if(currBasal.deliveryType === 'temp' && event.deliveryType === 'temp' && event.duration !== 0) {
+          // temp basal was updated
+          currBasal.with_expectedDuration(currBasal.duration)
+                   .with_duration(Date.parse(event.time) - Date.parse(currBasal.time));
+          event.suppressed = _.clone(currBasal.suppressed);
+        }
+
+        if(currBasal.deliveryType === 'temp' && currPumpSettings) {
+          var currSchedule = currPumpSettings.basalSchedules[currPumpSettings.activeSchedule];
+          var startTime = common.computeMillisInCurrentDay(currBasal);
+          var endTime = startTime + currBasal.duration;
+
+          for(var i in currSchedule) {
+            if((startTime < currSchedule[i].start) && (endTime > currSchedule[i].start)) {
+              // there was a schedule change _during_ the temp basal
+              var adjustedDuration = currSchedule[i].start - startTime;
+              var oldDuration = currBasal.duration;
+              currBasal.with_duration(adjustedDuration)
+                       .with_expectedDuration(oldDuration);
+              currBasal = currBasal.done();
+              events.push(currBasal);
+
+              var newJsDate = new Date(currBasal.jsDate.valueOf() + adjustedDuration);
+
+              var newSuppressed = {
+                type: 'basal',
+                deliveryType: 'scheduled',
+                rate: currSchedule[i].rate
+              };
+
+              var newTemp = config.builder.makeTempBasal()
+                .with_time(newJsDate.toISOString())
+                .with_duration(oldDuration - adjustedDuration)
+                .with_deviceTime(sundial.formatDeviceTime(newJsDate))
+                .set('index', currBasal.index)
+                .set('suppressed', newSuppressed); // TODO: build array of suppressed objects
```

Comment:
Not sure I understand your comment here. Are we changing the nested suppressed to an array of them?